homeassistant/components/light/__init__.py

```diff
@@ -689,7 +725,9 @@ async def async_initialize(self) -> None:
         self.data = await self.hass.async_add_executor_job(self._load_profile_data)
 
     @callback
-    def apply_default(self, entity_id: str, state_on: bool, params: dict) -> None:
+    def apply_default(
+        self, entity_id: str, state_on: bool | None, params: dict
```

Comment:
Is it possible to define the data types within the `dict`?

Reply:
`params` should be `dict[str, Any]`.

The specialization for `params` is also missing from `apply_profile` directly below.